### PR TITLE
Add Details Modal

### LINKED ISSUE #8: Create SessionDetails component
## Description
SessionDetails is a component that will take in a Session as a prop and will display the details of a session when a user clicks on its summary (SessionSummary) on the Studying Activity list.

## Learning objectives
- React
- API Wrapper
- Frontend

## Todo
- Create `SessionDetails.jsx` in `\client\src\components`
- Add logic and CSS to component
- Can test by adding to Home Page

### PR DESCRIPTION
**Pull request by exrlla**
## Status:
**last updated: 11/16/2021 9:41AM**
- modal component added 
- works onClick and displays a Session Summary 

Fixes #8 

## Todos

- join now button that toggles user to join a session 
- color and styling

## Screenshots
Screen Shot 2021-11-16 at 9 43 30 AM

### CODE REVIEW

File: client/src/css/Home.scss [hunk 1]
```diff
@@ -1 +1,82 @@
...
+.sessionCard {
+  box-shadow: none !important;
+  text-align: left !important;
+  width: 400px !important;
+  height: 100px !important;
+  margin-bottom: 20px !important;
+  border-radius: 15px !important;
+  transition: all 0.3s ease !important;
+  // box-shadow: 3px 3px 6px grey !important;
+}
```
Comment (dyang5200) [inline client/src/css/Home.scss]: Please extract all CSS related to the SessionSummary component into a separate SessionSummary.scss file!

File: client/src/css/Home.scss [hunk 2]
```diff
@@ -1 +1,82 @@
...
+
+.ui.card > .content > .header {
+  font-family: Tommy !important;
+}
+.sessionCard:hover {
+  transition: all 0.3s ease;
+  background-color: #b2caee !important;
+  box-shadow: 3px 3px 6px grey !important;
+}
```
Comment (dyang5200) [inline client/src/css/Home.scss]: The cool thing about Sass (.scss) is that you can nest CSS rules inside one another instead of having them all separate. So, for anything that is inside .sessionCard, you can nest their rules inside the .sessionCard field like this:

```
.sessionCard {
  box-shadow: none !important;
  text-align: left !important;
  width: 400px !important;
  height: 100px !important;
  margin-bottom: 20px !important;
  border-radius: 15px !important;
  transition: all 0.3s ease !important;
  // box-shadow: 3px 3px 6px grey !important;

  &:hover {
    transition: all 0.3s ease;
    background-color: #b2caee !important;
    box-shadow: 3px 3px 6px grey !important;
  }

  .insideCard {
    margin-top: 5px !important;
    padding: 20px !important;
    font-family: Tommy !important;
  }

  #join-btn {
    padding: 5px 27px;
    margin-left: 280px;
    transition: all 0.3s ease !important;
    font-family: Tommy !important;

    &:hover {
      transition: all 0.3s ease !important;
    }
  }
}
```

(I just copied a few fields, might have missed some, so make sure you nest all relevant fields inside .sessionCard)

File: client/src/css/Home.scss [hunk 3]
```diff
@@ -1 +1,82 @@
...
+  font-family: Tommy;
+  padding: 5px 27px;
+  margin-left: 280px;
+  transition: all 0.3s ease;
+}
+
+.homeHeader {
+  margin-top: 50px;
+  margin-left: 100px;
+  margin-bottom: 50px;
```
Comment (dyang5200) [inline client/src/css/Home.scss]: If you're setting multiple margins, you can set them in one line like this:

`  margin: 50px 0px 50px 100px; `

The syntax is margin: margin-top value, margin-right value, margin-bottom value, margin-left value. (ie - starts from margin-top and goes clockwise)

File: client/src/css/Home.scss [hunk 4]
```diff
@@ -1 +1,82 @@
...
+.ui.modal > .content {
+  background: #c5cad4;
+  padding-left: 25px;
+  padding-bottom: 30px;
+}
+
+.ui.modal > :last-child {
+  border-bottom-left-radius: 100px;
+  border-bottom-right-radius: 100px;
+}
```
Comment (dyang5200) [inline client/src/css/Home.scss]: Nest these rules as well, and extract them (and other relevant css rules) into a separate DetailsModal.scss

File: client/src/components/modals/DetailsModal.jsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+            Class: {session.class}
+            <br />
+            Location: {session.location}
+            <br />
+            Expected Group Size: {session.attendees}
+            <br />
+            Notes from {session.creator}
+            <div className="notes-bg">
+              <p>{session.notes}</p>
+            </div>
```
Comment (dyang5200) [inline client/src/components/modals/DetailsModal.jsx]: Can you add the startTime to this too? Make sure to convert it from [epoch time](https://www.epochconverter.com/) to a Date

File: client/src/components/modals/DetailsModal.jsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    >
+      <Modal.Content>
+        <Modal.Description>
+          <Header>Join {session.creator}</Header>
+          <p id="form-text">
+            Class: {session.class}
+            <br />
+            Location: {session.location}
+            <br />
+            Expected Group Size: {session.attendees}
```
Comment (dyang5200) [inline client/src/components/modals/DetailsModal.jsx]: In the [database](https://docs.google.com/document/d/17DP6zekcG-BVo69tBjTnzAj5OhXQ5b3O8aey9nKG-jE/edit?usp=sharing), `attendees` is a string array. Please replace group size to a list of all attendees. Even though you're currently working with fake data, please make sure the types are consistent with the types in the database to avoid errors in the future once we start using real data!

**Review by dyang5200 — CHANGES_REQUESTED** [review]
Ellie + Aaron -- this looks SO cool! You two are CSS gods 🥇 Just a couple code style comments! Let me know if you guys have any questions

File: client/src/components/modals/DetailsModal.jsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+            <br />
+            <br />
+            Notes from {session.creator}:
+            <div className="notes-bg">
+              <div className="ui form">
+                <div className="fifteen wide field">
+                  <textarea
+                    rows="3"
+                    placeholder="How's the study sesh going?"
+                  ></textarea>
```
Comment (ashwin1104) [inline client/src/components/modals/DetailsModal.jsx]: Why is this an input field? Shouldn't "notes from the session creator" already exist? These notes should not be editable, they should simply be displayed.

File: client/src/components/modals/DetailsModal.jsx [hunk 8]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
```
Comment (ashwin1104) [inline client/src/components/modals/DetailsModal.jsx]: You guys did an impeccable job with this modal component. This is a quality implementation right here 🔥

File: client/src/css/DetailsModal.scss [hunk 9]
```diff
@@ -0,0 +1,37 @@
+@import '../styles/colors.scss';
+
+.ui.modal {
+  background-color: #c5cad4 !important;
+  font-family: Tommy;
+  width: 500px;
+
+  .ui.header:not(h1):not(h2):not(h3):not(h4):not(h5):not(h6) {
```
Comment (ashwin1104) [inline client/src/css/DetailsModal.scss]: Why did you need to add all these not(h*) selectors to the .ui.header rule?

File: client/src/css/Home.scss [hunk 10]
```diff
@@ -1 +1,44 @@
...
+  transition: all 0.3s ease !important;
+}
+
+.homeHeader {
+  margin: 50px 0px 50px 100px;
+  font-size: 80px;
+  text-align: left;
+  font-family: Tommy;
+}
+h1 {
```
Comment (ashwin1104) [inline client/src/css/Home.scss]: replace this with a classname

File: client/src/components/modals/DetailsModal.jsx [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+import { Button, Modal, Header } from 'semantic-ui-react';
+import 'semantic-ui-css/semantic.min.css';
+
```
Comment (ashwin1104) [inline client/src/components/modals/DetailsModal.jsx]: import the detailModal styles in this component, not in the homepage

File: client/src/components/SessionSummary.jsx [hunk 12]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { Card } from 'semantic-ui-react';
+
```
Comment (ashwin1104) [inline client/src/components/SessionSummary.jsx:5]: import the SessionSummary styles in this component, not in the homepage

File: client/src/pages/Home.jsx [hunk 13]
```diff
@@ -1,10 +1,32 @@
 import React, { useEffect, useState } from 'react';
 
+import SessionSummary from '../components/SessionSummary';
+// import HomeHeader from '../components/HomeHeader';
 import { getSampleResponse } from '../utils/apiWrapper';
-
 import '../css/Home.scss';
+import '../css/SessionSummary.scss';
+import '../css/DetailsModal.scss';
```
Comment (ashwin1104) [inline client/src/pages/Home.jsx]: Remove the SessionSummary and DetailModal CSS imports. They should instead be imported in their respective component files.

File: client/src/styles/styles.scss [hunk 14]
```diff
@@ -1,5 +1,10 @@
 @import 'colors.scss';
 
+@font-face {
+  font-family: 'Tommy';
+  src: local('Tommy'), url('../fonts/Tommy.ttf') format('truetype');
+}
+
 body {
   background-color: $light;
   color: $primary;
```
Comment (ashwin1104) [inline client/src/styles/styles.scss]: Add `font-family: Tommy` to this body rule so that every page in our app uses this font. If you do this, you shouldn't need to add `font-family: Tommy` to any other CSS file/rule (which means you can take it out of the SessionSummary and DetailsModal CSS files)

File: package.json [hunk 15]
```diff
@@ -0,0 +1,6 @@
+{
+  "dependencies": {
+    "bootstrap": "^5.1.3",
+    "react-bootstrap": "^2.0.2"
```
Comment (ashwin1104) [inline package.json]: Remove this file. You probably accidentally ran `yarn` in the parent directory of our repo, instead of in client/.

Comment (ashwin1104) [reply package.json]: Remove the package-lock.json file as well.

File: client/src/pages/Home.jsx [hunk 16]
```diff
@@ -1,10 +1,32 @@
 import React, { useEffect, useState } from 'react';
 
+import SessionSummary from '../components/SessionSummary';
+// import HomeHeader from '../components/HomeHeader';
```
Comment (ashwin1104) [inline client/src/pages/Home.jsx]: Remove this comment

File: client/src/css/SessionSummary.scss [hunk 17]
```diff
@@ -0,0 +1,12 @@
...
+
+.sessionCard {
+  box-shadow: none !important;
+  text-align: left !important;
+  width: 400px !important;
+  height: 100px !important;
+  margin-bottom: 20px !important;
+  border-radius: 15px !important;
+  transition: all 0.3s ease !important;
+  // box-shadow: 3px 3px 6px grey !important;
```
Comment (ashwin1104) [inline client/src/css/SessionSummary.scss]: Remove comment

File: client/src/css/Home.scss (hunk 10 above)
Comment (ashwin1104) [reply client/src/css/Home.scss]: Or get rid of it, doesn't look like you need it

**Review by ashwin1104 — CHANGES_REQUESTED** [review]
You guys tackled a challenging frontend feature and have done an amazing job with it. This is beautiful - super proud of what you've created here :) I requested some important changes, please address ASAP.

**Review by dyang5200 — APPROVED** [review]
Amazing work guys! ❤️ Super impressed

File: client/src/pages/Home.jsx [hunk 18]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 
 import SessionSummary from '../components/SessionSummary';
-
+// import { getSampleResponse } from '../utils/apiWrapper';
```
Comment (ashwin1104) [inline client/src/pages/Home.jsx]: Remove comment

File: client/src/components/SessionSummary.jsx [hunk 19]
```diff
@@ -1,5 +1,7 @@
 import React from 'react';
 import { Card } from 'semantic-ui-react';
+
+import DetailsModal from './DetailsModal';
```
Comment (ashwin1104) [inline client/src/components/SessionSummary.jsx]: import the SessionSummary.scss file here

**Review by ashwin1104 — CHANGES_REQUESTED** [review]
Beautiful work!! Y'all killed this feature 🔥👏🏼 just requested a couple minor changes, otherwise it's ready to merge!
